### ISSUE TITLE
Add key for BottomNavigationBarItem

### ISSUE THREAD

**Issue by JoeCiou** [flutter/flutter#34833]
I have check this link:
https://github.com/flutter/flutter/blob/master/packages/flutter/test/material/bottom_navigation_bar_test.dart

It use `tester.tap` to drive one of BottomNavigationBarItem. but It is not very convenient if app support localization.
I think it add key can more flex to test.

**Comment by LongCatIsLooong** [comment]
I don't think those tests are meant to be localized. The strings are just hard coded code names given to the widgets in order to be able to identify them later in the test. I'm going to close this issue but feel free to reopen if I missed anything.

**Comment by JoeCiou** [comment]
Assume the code like this: 
```
String localizationTabTitleString(int index) {
// get different title from specific language
}
```
```
bottomNavigationBar: BottomNavigationBar(
  items: const <BottomNavigationBarItem>[
    BottomNavigationBarItem(
      icon: Icon(Icons.ac_unit),
      title: Text(localizationTabTitleString(0)),
    ),
    BottomNavigationBarItem(
      icon: Icon(Icons.access_alarm),
      title: Text(localizationTabTitleString(1)),
    ),
  ],
),
```

then I must use `localizationTabTitleString` to test:
```
await tester.tap(find.text(Text(localizationTabTitleString(0)));
```

I think this is not make sense? or do I miss other way to write BottomNavigationBar test?

**Comment by JoeCiou** [comment]
Feel embarrassed. I should use `flutter_driver` to make sense. 
Ref this [link](https://stackoverflow.com/questions/55460993/flutter-driver-test-bottomnavigationbaritem).

**Comment by LongCatIsLooong** [comment]
Ah sorry, I misunderstood the problem. If you're writing tests for your app, feel free to use keys for widget lookup.

**Comment by gronka** [comment]
I agree with @JoeCiou. It's unintuitive and impedes testing to omit a key for a random widget (BottomNavigationBarItem). Is there a reason for this? If not, @LongCatIsLooong I'd like the issue reopened and I could open a PR.

ref: https://api.flutter.dev/flutter/widgets/BottomNavigationBarItem-class.html

**Comment by LongCatIsLooong** [comment]
@gronka  Ah sorry looks like I read the issue wrong. Like you said `BottomNavigationBarItem` is not a widget but I'm reopening the issue in case you're still interested in opening a PR. 

Not sure why exactly it's made that way, the class seems to be used by both the material bottom nav bar class and the Cupertino navbar class, so it's possible that it's intentionally made a non-widget so it can stay design language agnostic.

 /cc @HansMuller

**Comment by madscs11** [comment]
Apologies for commenting on an older issue but, as it's still open, I'd just like to add that there is a different use-case where this is unintuitive.

I'm currently implementing [coach marks](https://pub.dev/packages/tutorial_coach_mark) and I'd like to add a coach mark to the item itself but I'd either have to focus on the icon or implement a custom bottom navigation bar myself.

**Comment by HansMuller** [comment]
@gronka - BottomNavigationBarItem is used to produce a single widget in the Material BottomNavigationBar and CupertinoTabBar implementations, so providing and eventually using a key would make sense. This isn't something we're working on at the moment but happy to review a PR.

**Comment by Gibbo97** [comment]
We are having an issue that would be solved by this.

We dynamically change the number of BottomNavigationBarItem's we have in the bar depending on what one you currently have selected. This causes the splash / inkwell effect to happen on the wrong item, the one adjacent to the one you tapped. Adding keys fixes this.

**Comment by ShakilKhan999** [comment]
> Apologies for commenting on an older issue but, as it's still open, I'd just like to add that there is a different use-case where this is unintuitive.
> 
> I'm currently implementing [coach marks](https://pub.dev/packages/tutorial_coach_mark) and I'd like to add a coach mark to the item itself but I'd either have to focus on the icon or implement a custom bottom navigation bar myself.

can you please provide the code?
